File: src/main/java/cube8540/oauth/authentication/credentials/authority/domain/SecuredResourceRepository.java
package cube8540.oauth.authentication.credentials.authority.domain;

import org.springframework.data.jpa.repository.EntityGraph;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;

public interface SecuredResourceRepository extends JpaRepository<SecuredResource, SecuredResourceId> {

    Long countByResourceId(SecuredResourceId resourceId);

    @Override
    @EntityGraph(attributePaths = {"authorities"}, type = EntityGraph.EntityGraphType.LOAD)
    List<SecuredResource> findAll();

    @Override
    @EntityGraph(attributePaths = {"authorities"}, type = EntityGraph.EntityGraphType.LOAD)
    Optional<SecuredResource> findById(SecuredResourceId id);

}

File: src/main/kotlin/cube8540/oauth/authentication/resource/domain/SecuredResourceRepository.kt
package cube8540.oauth.authentication.resource.domain

import java.util.Optional
import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.JpaRepository

interface SecuredResourceRepository: JpaRepository<SecuredResource, SecuredResourceId> {

    fun countByResourceId(resourceId: SecuredResourceId): Long

    override fun findAll(): MutableList<SecuredResource>

    @EntityGraph(attributePaths = ["authorities"], type = EntityGraph.EntityGraphType.LOAD)
    override fun findById(id: SecuredResourceId): Optional<SecuredResource>

}